### PR TITLE
Fix animations with zero runtime length to give a useful error instead of a broken pipe

### PR DESCRIPTION
**Pull request by MrDiver**
Throwing an error when an animation group is instantiated with an empty list.

### CODE REVIEW

**Review by JasonGrace2282 — APPROVED** [review]
LGTM! Is it worth adding a test for this?

**Comment by behackl** [comment]
The issue actually sits a bit deeper. This happens for *any* animation as soon as the `run_time` is set to 0:

```py
anim = Animation(None, run_time=0)
self.play(anim)
self.wait()
```

also breaks rendering.

I think it is fine that we raise an error in this case (I can't really think of an example for a animation with 0 run time) -- but I'd rather do it in `Animation.begin` and `AnimationGroup.begin` after a check whether `self.run_time == 0`.

**Comment by MrDiver** [comment]
Does that interfere with your change of animation group?

File: manim/animation/animation.py
Comment (JasonGrace2282) [inline manim/animation/animation.py:1]: Might be better to change this to ``self.run_time >= 0``

File: manim/animation/composition.py
Comment (JasonGrace2282) [inline manim/animation/composition.py:1]: See previous.

**Review by JasonGrace2282 — CHANGES_REQUESTED** [review]

File: manim/animation/animation.py
Comment (MrDiver) [reply manim/animation/animation.py:1]: Do you mean `<= 0`

Comment (JasonGrace2282) [reply manim/animation/animation.py:1]: Ah yes my bad

File: manim/animation/animation.py [hunk 1]
```diff
@@ -191,6 +191,10 @@ def begin(self) -> None:
         method.
 
         """
+        if self.run_time == 0:
```
Comment (MrDiver) [inline manim/animation/animation.py:194]:
```suggestion
        if self.run_time <= 0:
```

File: manim/animation/composition.py [hunk 2]
```diff
@@ -81,6 +81,15 @@ def get_all_mobjects(self) -> Sequence[Mobject]:
         return list(self.group)
 
     def begin(self) -> None:
+        if self.run_time == 0:
```
Comment (MrDiver) [inline manim/animation/composition.py:84]:
```suggestion
        if self.run_time <= 0:
```

File: manim/animation/animation.py [hunk 3]
```diff
@@ -191,6 +191,10 @@ def begin(self) -> None:
         method.
 
         """
+        if self.run_time <= 0:
+            raise ValueError(
+                f"{self} has a runtime of 0 seconds. Which cannot be rendered correctly! please set a runtime > 0"
```
Comment (JasonGrace2282) [inline manim/animation/animation.py:196]:
```suggestion
                f"{self} has a runtime of <= 0 seconds, which cannot be rendered correctly! please set a runtime > 0"
```

File: manim/animation/composition.py [hunk 4]
```diff
@@ -81,6 +81,15 @@ def get_all_mobjects(self) -> Sequence[Mobject]:
...
 
     def begin(self) -> None:
+        if self.run_time <= 0:
+            tmp = (
+                "please set a runtime > 0"
+                if len(self.animations) != 0
+                else "Please add at least one Animation"
+            )
+            raise ValueError(
+                f"{self} has a runtime of 0 seconds. Which cannot be rendered correctly! {tmp}."
```
Comment (JasonGrace2282) [inline manim/animation/composition.py:91]:
```suggestion
                f"{self} has a runtime of <= 0 seconds, which cannot be rendered correctly! please set a runtime > 0"
```

File: tests/module/animation/test_composition.py [hunk 5]
```diff
@@ -169,3 +169,13 @@ def test_animationgroup_is_passing_remover_to_nested_animationgroups():
     assert sqr_animation.remover
     assert circ_animation.remover
     assert polygon_animation.remover
+
+
+def test_empty_animation_group_fails():
+    with pytest.raises(ValueError):
+        AnimationGroup().begin()
```
Comment (JasonGrace2282) [inline tests/module/animation/test_composition.py:176]: Maybe check the text of the error?

**Review by JasonGrace2282 — COMMENTED** [review]
Seems pretty good, just some minor suggestions maybe?

File: manim/animation/composition.py (hunk 4 above)
Comment (MrDiver) [reply manim/animation/composition.py:91]: Why exactly should we remove the differentiation between 0 runtime and no animations for the animation group ? They are two distinct errors.

File: tests/module/animation/test_composition.py (hunk 5 above)
Comment (MrDiver) [reply tests/module/animation/test_composition.py:176]: Maybe

File: manim/animation/composition.py (hunk 4 above)
Comment (JasonGrace2282) [reply manim/animation/composition.py:91]: Oh, ignore that last part, copy-paste error 😅

**Comment by MrDiver** [comment]
# No Font List Check
## LRU
grafik

## Manual

grafik

## Main

grafik

**Comment by MrDiver** [comment]
grafik

**Review by JasonGrace2282 — APPROVED** [review]

File: tests/module/animation/test_composition.py (hunk 5 above)
Comment (MrDiver) [reply tests/module/animation/test_composition.py:176]: Actually i don't think we need to do that, the ValueError should be specific enough in that case. As long as it gives an error at that stage where good to go

**Review by behackl — COMMENTED** [review]
Welp, just noticed I had started to review this at some point; error messages could have been written in a more consistent way.

Anyhow, not that important.

**Comment by MrDiver** [comment]
I will move over your changes to an extra commit on main!

**Pull request merged**